e2e_test.go

```diff
@@ -25,6 +25,12 @@ func TestE2EFullTopology(t *testing.T) {
 		"./examples/basic/", "", "", "",
 	))
 
+	t.Run("validation", testE2EFullTopology(
+		"./examples/validation/", "",
+		"_out/output.csv.gz",
+		"testdata/output.csv.gz.golden",
+	))
+
```

Comment:
This example doesn't really illustrate nor test the new validation feature. Both files (input and output) contain:

```csv
value1,value2,value3
value1,value2,value3
value5,value2,value3
```
so no record has been invalidated by the validation function.

Here we could for example use an already existing input file, say `testdata/input.csv.zst`, but make it so that not all records pass the validation function generated from `validation/main.go`.
At this point the output file will be different, so you can add new `golden` file, `./testdata/output.validation.csv.gz.golden` for example, into the repository, and update the end-to-end with that new file

Reply:
The input file of the example, i.e. `testdata/input.csv.zst`, contains:
```
value1,value2,value3
value1,value2,value3
value4,value2,value6
value5,value2,value3
```
Therefore the validation remove the record `(value4,value2,value6)`. I think that the example correctly illustrates the feature: the `timestamp` could be any `value+a number`, `source` is not validated and `target` should contain `value3`.
Anyway, if you think that the example should be more clear I am open to suggestions.

Moreover, I take your suggestion and I will modify the output file from `output.csv.gz.golden` in `output.validation.csv.gz.golden`.

Reply:
my bad I opened the wrong file. :+1: